lib/vector-core/src/enrichment_table/mod.rs

```diff
@@ -0,0 +1,17 @@
+pub mod enrichment_tables;
+
+pub use enrichment_tables::EnrichmentTables;
+
+use std::collections::BTreeMap;
+
+/// Enrichment tables represent additional data sources that can be used to enrich the event data
+/// passing through Vector.
+pub trait EnrichmentTable: std::fmt::Debug {
+    /// Search the enrichment table data with the given condition.
+    /// All fields within the data much match (AND).
+    fn find_table_row(&self, condition: BTreeMap<String, String>) -> Option<&Vec<String>>;
```

Comment:
Could we change this into a `Vec<Condition>`? Right now, it's not entirely clear what the key and value part of the map actually represent in a condition.

It would also help with future extensions to the conditions (although that's not the primary motivation for requesting this change).